React/Views/RCTSliderManager.m

```diff
@@ -30,6 +30,13 @@ - (UIView *)view
 
 static void RCTSendSliderEvent(RCTSlider *sender, BOOL continuous)
 {
+  if (sender.step > 0 &&
+    sender.step <= (sender.maximumValue - sender.minimumValue)) {
+    int step = ((int)((sender.value + sender.step / 2) / sender.step) * sender.step);
+    [sender setValue: step
+            animated: NO];
+  }
+
   if (sender.onChange) {
     sender.onChange(@{
       @"value": @(sender.value),
```

Comment:
We should avoid sending events when the value hasn't actually changed. Currently this would send an event each time the slider was dragged, even if the stepped value rounded to the same amount.